### PR TITLE
Implement test_setting_toleration_extra

### PR DESCRIPTION
**Pull request by c3y1huang**
https://github.com/longhorn/longhorn/issues/1977

### CODE REVIEW

File: manager/integration/tests/test_settings.py [hunk 1]
```diff
@@ -183,26 +191,87 @@ def test_setting_toleration_extra():
...
+        },
+        {
+            "value": "",
+            "expect": [],
+        },
+    ]
+
+    chk_removed_tolerations = []
+    for setting in settings:
+        client = get_longhorn_api_client()  # NOQA
```
Comment (PhanLe1010) [inline manager/integration/tests/test_settings.py:242]: Why do you put this one inside the `for` loop?

File: manager/integration/tests/test_settings.py [hunk 2]
```diff
@@ -183,26 +191,87 @@ def test_setting_toleration_extra():
...
+        taint_toleration = client.by_id_setting(SETTING_TAINT_TOLERATION)
+        updated = client.update(taint_toleration,
+                                value=setting["value"])
+        assert updated.value == setting["value"]
+
+        core_api = get_core_api_client()  # NOQA
```
Comment (PhanLe1010) [inline manager/integration/tests/test_settings.py:248]: Why do you put this one inside the `for` loop?

File: manager/integration/tests/test_settings.py [hunk 3]
```diff
@@ -183,26 +191,87 @@ def test_setting_toleration_extra():
...
         for p in pod_list:
-            if p.status.phase != "Running" or \
-                    not check_tolerations_set(p.spec.tolerations,
-                                              set_tolerations):
-                updated = False
+            if p.status.phase == "Running" \
+                and check_tolerations_set(p.spec.tolerations,
+                                          expected_tolerations,
+                                          chk_removed_tolerations):
+                updated = True
```
Comment (PhanLe1010) [inline manager/integration/tests/test_settings.py:274]: I think there is an issue here. The expectation is that all pods in `longhorn-system` namespace must be updated. However, the new code is reporting ok if there is a pod that was updated. How do you think?

File: manager/integration/tests/test_settings.py (hunk 2 above)
Comment (c3y1huang) [reply manager/integration/tests/test_settings.py]: You are right, it's not required. Addressed.

File: manager/integration/tests/test_settings.py (hunk 3 above)
Comment (c3y1huang) [reply manager/integration/tests/test_settings.py]: You are correct. Addressed.

File: manager/integration/tests/test_settings.py (hunk 1 above)
Comment (c3y1huang) [reply manager/integration/tests/test_settings.py:242]: Seems the update triggers pods to restart and have a new cluster IP assignment. Hence makes the previous client invalid.

**Comment by c3y1huang** [comment]
@PhanLe1010 thanks for your review, all comments are addressed.

File: manager/integration/tests/test_settings.py (hunk 1 above)
Comment (PhanLe1010) [reply manager/integration/tests/test_settings.py:242]: Thank you for the explanation!

**Review by PhanLe1010 — APPROVED** [review]
LGTM

**Pull request merged**